application/src/Utils.ts

```diff
@@ -1,71 +1,25 @@
-import { Point } from "chart.js";
-
-export const getServerHostname = () => {
-  return location.host;
-};
-
+import Inputs from "./scouter/Inputs";
+import ScouterInput from "./scouter/ScouterInput";
 
 export function rangeArr(rangeStart: number, rangeEnd: number): number[] {
-  return Array.from({length:rangeEnd - rangeStart})
-    .map((_, i) => i + rangeStart);
-}
-
-export function matchToSheet(match: Record<string, string>) {
-  let keys = "";
-  let values = "";
-  for (const [key, value] of Object.entries(match)) {
-    keys += key + "	";
-    values += value + "	";
-  }
-  return `${keys}\n${values}`;
-}
-
-export async function getMatchesByCriteria(field?: string, value?: string) {
-  const searchedField = field && value ? `/${field}/${value}` : ``;
-  const data: Record<string, string>[] = await fetch(
-    `https://${getServerHostname()}/Matches${searchedField}`,
-    {
-      method: "GET",
-      mode: "cors",
-      headers: {
-        "Content-Type": "application/json",
-      },
-    }
-  )
-    .then((response) => {
-      if (!response.ok) {
-        throw new Error("Network response was not ok");
-      }
-      return response.json();
-    })
-    .catch((error) => {
-      console.log(error);
-      return [];
-    });
-  return data;
-}
-
-export function sortMatches(matches: Match[]) {
-  return matches.sort(
-    (match1, match2) => {
-      return parseInt(match1["Qual"]) - parseInt(match2["Qual"]);
-        }
+  return Array.from({ length: rangeEnd - rangeStart }).map(
+    (_, i) => i + rangeStart
   );
 }
 
-export type Match = Record<string, string>;
-export interface Note extends Point {
-  color: "green" | "red" | "orange";
+export function sortMatches(matches: Match[]) {
+  return matches.sort((match1, match2) => match1.Qual - match2.Qual);
 }
 
-export const autoNotePositions: Point[] = [250, 200, 150, 100, 50].map(
-  (height) => {
-    return { x: 280, y: height };
-  }
-);
-export const autoBlueNotePositions: Point[] = [150, 100, 50].map((height) => {
-  return { x: 90, y: height };
-});
+type InputName = keyof typeof Inputs;
+type ExcludeByType<T, U> = {[K in keyof T as T[K] extends U ? never : K]: T[K];};
+type InputsMapped = {
+  // ScouterName: string, Qual: number ...
+  [K in InputName]: (typeof Inputs)[K] extends ScouterInput<infer U, any, any>
+    ? U
+    : never;
+};
+export type Match = ExcludeByType<InputsMapped, never>;
 
 export const FRCTeamList = [
```

Comment:
What?

Reply:
? its the teams in frc, which are used in the table where all of their data is shown.
in the future we will use data from the TBA to get the teams in a specific event.